src/treeview/treeview.component.ts

```diff
@@ -57,7 +57,8 @@ import { TreeViewService } from "./treeview.service";
 			<ng-template #notProjected>
 				<cds-tree-node
 					*ngFor="let node of tree"
-					[node]="node">
+					[node]="node"
+					(nodetoggle)="onNodeToggle($event)">
```

Comment:
Oh right, similar to the focus node i had to do in the other PR....anyway yes feel free to push directly to the PR.

About renaming it would make more sense indeed, or even better `expandedChange` so that it could be also used the 2 way binding

Reply:
actually on a second thought having an observable in the service is not needed. Listening for `nodeToggle` from the top component only makes sense if we don't use projected content, otherwise the user would just listen on the nodeComponent itself. So i believe we just need to add the listener to `nodetoggle` in the nodeComponent itself for its children and propagate that value through the parent output. (Let me know if it's not clear from my explantation here...i'll try to phrase it better with an example)